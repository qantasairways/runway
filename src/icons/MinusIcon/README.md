```js
<MinusIcon width="50" height="50" />
```
