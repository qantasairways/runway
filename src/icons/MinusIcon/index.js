/* eslint-disable react/no-unknown-property */
import React from 'react';
import PropTypes from 'prop-types';

import { generate } from 'shortid';

const MinusIcon = ({ className, width, height, ...rest }) => {
  const pathId = generate();
  const maskId = generate();
  return (
    <svg
      width={width}
      height={height}
      className={className}
      viewBox="0 0 24 24"
      focusable="false"
      {...rest}
    >
      <defs>
        <path
          id={pathId}
          d="M17.214 13.25H7.786c-.432 0-.786-.338-.786-.75s.354-.75.786-.75h9.428c.432 0 .786.338.786.75s-.354.75-.786.75z"
        />
      </defs>
      <g fillRule="evenodd">
        <mask id={maskId} fill="#fff">
          <use xlinkHref={`#${pathId}`} />
        </mask>
        <g mask={`url(#${maskId})`}>
          <path d="M0 0h24v24H0z" />
        </g>
      </g>
    </svg>
  );
};

PropTypes.propTypes = {
  className: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string
};

export default MinusIcon;
