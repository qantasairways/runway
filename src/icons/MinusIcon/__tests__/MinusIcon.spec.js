import React from 'react';
import { shallow } from 'enzyme';
import shortid from 'shortid';

import MinusIcon from '..';

jest.mock('shortid');

describe('MinusIcon', () => {
  it('should render', () => {
    shortid.generate.mockReturnValue('1a2b3c');
    const component = shallow(
      <MinusIcon width="36px" height="36px" className="my-favourite-icon" />
    );

    expect(component).toMatchSnapshot();
  });

  it('should set the width', () => {
    const component = shallow(<MinusIcon width="36px" />);
    const result = component.find('svg').props().width;

    expect(result).toEqual('36px');
  });

  it('should set the height', () => {
    const component = shallow(<MinusIcon height="36px" />);
    const result = component.find('svg').props().height;

    expect(result).toEqual('36px');
  });

  it('should set the class name', () => {
    const component = shallow(<MinusIcon className="my-favourite-icon" />);
    const result = component.find('svg').props().className;

    expect(result).toEqual('my-favourite-icon');
  });
});
