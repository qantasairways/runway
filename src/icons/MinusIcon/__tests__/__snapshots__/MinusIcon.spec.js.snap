// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinusIcon should render 1`] = `
<svg
  className="my-favourite-icon"
  focusable="false"
  height="36px"
  viewBox="0 0 24 24"
  width="36px"
>
  <defs>
    <path
      d="M17.214 13.25H7.786c-.432 0-.786-.338-.786-.75s.354-.75.786-.75h9.428c.432 0 .786.338.786.75s-.354.75-.786.75z"
      id="1a2b3c"
    />
  </defs>
  <g
    fillRule="evenodd"
  >
    <mask
      fill="#fff"
      id="1a2b3c"
    >
      <use
        xlinkHref="#1a2b3c"
      />
    </mask>
    <g
      mask="url(#1a2b3c)"
    >
      <path
        d="M0 0h24v24H0z"
      />
    </g>
  </g>
</svg>
`;
